DirectProgramming/C++SYCL_FPGA/Tutorials/Features/multi_aocx/device_code_split/src/simple.cpp

```diff
@@ -0,0 +1,85 @@
+#include <sycl/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <iostream>
+
+using namespace sycl;
+constexpr access::mode sycl_read = access::mode::read;
+constexpr access::mode sycl_write = access::mode::write;
+
+class SimpleAdd;
+class SimpleMul;
+
+void kernel_vadd(const float a, const float b, float &c) {
+
+  {
+#if FPGA_SIMULATOR
+    auto selector = sycl::ext::intel::fpga_simulator_selector_v;
+#elif FPGA_HARDWARE
+    auto selector = sycl::ext::intel::fpga_selector_v;
+#else // #if FPGA_EMULATOR
+    auto selector = sycl::ext::intel::fpga_emulator_selector_v;
+#endif
+
+    queue deviceQueue(selector);
+    buffer<cl_float, 1> bufferA(&a, 1);
+    buffer<cl_float, 1> bufferB(&b, 1);
+    buffer<cl_float, 1> bufferC(&c, 1);
+
+    deviceQueue.submit([&](handler &cgh) {
+      auto const accessorA = bufferA.template get_access<sycl_read>(cgh);
+      auto accessorB = bufferB.template get_access<sycl_read>(cgh);
+      auto accessorC = bufferC.template get_access<sycl_write>(cgh);
+      cgh.single_task<class SimpleAdd>(
```

Comment:
```suggestion
      cgh.single_task<SimpleAdd>(
```
this class has been forward declared above so no need to add class keyword here.